kpops/components/streams_bootstrap/model.py

```diff
@@ -19,23 +28,207 @@
+    :param configuration_env_prefix: Prefix for environment variables to use that should be parsed as command line arguments.
+    :param command_line: Map of command line arguments passed to the streams app.
+    :param env: Custom environment variables.
+    :param secrets: Custom secret environment variables. Prefix with configurationEnvPrefix in order to pass secrets to command line or prefix with KAFKA_ to pass secrets to Kafka Streams configuration.
+    :param secret_refs: Inject existing secrets as environment variables. Map key is used as environment variable name. Value consists of secret name and key.
+    :param secret_files_refs: Mount existing secrets as volumes
+    :param files: Map of files to mount for the app. File will be mounted as $value.mountPath/$key. $value.content denotes file content (recommended to be used with --set-file).
+    :param pod_annotations: Map of custom annotations to attach to the pod spec.
+    :param pod_labels: Map of custom labels to attach to the pod spec.
+    :param liveness_probe: See https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.25/#probe-v1-core
+    :param readiness_probe: See https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.25/#probe-v1-core
+    :param affinity: Map to configure pod affinities https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#affinity-and-anti-affinity.
+    :param tolerations: Array containing taint references. When defined, pods can run on nodes, which would otherwise deny scheduling.
     """
 
+    image: str = Field(
+        description=describe_attr("image", __doc__),
+    )
+
     image_tag: str = Field(
         default="latest",
         pattern=IMAGE_TAG_PATTERN,
         description=describe_attr("image_tag", __doc__),
     )
 
+    image_pull_policy: ImagePullPolicy = Field(
+        default=ImagePullPolicy.ALWAYS,
+        description=describe_attr("image_pull_policy", __doc__),
+    )
+
+    image_pull_secret: list[dict[str, str]] = Field(
+        default_factory=list,
+        description=describe_attr("image_pull_secret", __doc__),
+    )
+
     kafka: KafkaConfig = Field(
         description=describe_attr("kafka", __doc__),
     )
 
+    resources: Resources = Field(
+        default=Resources(
+            requests=ResourceRequests(cpu="100m", memory="500Mi"),
+            limits=ResourceLimits(cpu="300m", memory="2G"),
+        ),
+        description=describe_attr("resources", __doc__),
+    )
+
+    ports: list[PortConfig] = Field(
+        default_factory=list,
+        description=describe_attr("ports", __doc__),
+    )
+
+    service: ServiceConfig = Field(
+        default_factory=ServiceConfig,
+        description=describe_attr("service", __doc__),
+    )
+
+    configuration_env_prefix: str = Field(
+        default="APP",
+        description=describe_attr("configuration_env_prefix", __doc__),
+    )
+
+    command_line: dict[str, Any] = Field(
```

Comment:
shouldn't the command line values be strings? or at least primitives like `str | bool | int`?

Reply:
I was not 100% sure about it. But I think the triple you mentioned here makes sense.

Reply:
I don't think so tbh. also floats are a slippery slope, I would prefer using decimals as strings instead to avoid rounding errors